data_structures/linked-list/linked_list.py

```diff
@@ -41,66 +41,74 @@ def find(self, val):
 
     def append(self, val):
         """Append a node to the end of the list."""
-        if val is str:
-            raise Exception('Please enter an integer, not a string!')
-
-        if self.head is None:
-            self.insert(val)
-        else:
-            search = self.head
-            while search:
-                if search._next is None:
-                    search._next = Node(val)
-                    self._size += 1
-                    break
-                search = search._next
-
+        if type(val) is not int:
+            raise Exception('Please enter an integer.')
```

Comment:
The data type of the nodes is not restricted in the spec, so this is not necessary.

Reply:
If you did want to raise an error, make sure that you are raising more specific exceptions.